storage/core/src/main/java/com/hortonworks/streamline/storage/impl/jdbc/JdbcStorageManager.java

```diff
@@ -61,14 +63,17 @@ public JdbcStorageManager(QueryExecutor queryExecutor) {
     @Override
     public void add(Storable storable) throws AlreadyExistsException {
         log.debug("Adding storable [{}]", storable);
-        final Storable existing = get(storable.getStorableKey());
-
-        if(existing == null) {
-            addOrUpdate(storable);
-        } else if (!existing.equals(storable)) {
-            throw new AlreadyExistsException("Another instance with same id = " + storable.getPrimaryKey()
-                    + " exists with different value in namespace " + storable.getNameSpace()
-                    + ". Consider using addOrUpdate method if you always want to overwrite.");
+
+        try {
+            queryExecutor.insert(storable);
+        } catch (StorageException ex) {
+            if (ex.getCause() instanceof SQLIntegrityConstraintViolationException) {
+                throw new AlreadyExistsException("Another instance with same id = " + storable.getPrimaryKey()
```

Comment:
I would prefer to go with right behavior. We can add more error logging about the error details which may be helpful in finding the root cause.